src/main/webapp/app/exercises/modeling/participate/modeling-submission.component.ts

```
@@ -132,23 +150,33 @@
-                        next: (modelingSubmission) => {
+            this.route.params
+                .pipe(
+                    switchMap((params) => {
+                        this.participationId = params['participationId'] ?? this.participationId;
+                        this.submissionId = Number(params['submissionId']) || undefined;
+                        this.isFeedbackView = !!this.submissionId;
+
+                        // If participationId exists and feedback view is needed, fetch history results first
+                        if (this.participationId && this.isFeedbackView) {
+                            return this.fetchSubmissionHistory().pipe(
+                                tap((results) => (this.sortedSubmissionHistory = results)),
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)